src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -295,213 +333,159 @@ public void shouldNotUpdateServiceIfConfidential() {
 
-        Phone newPhone = new Phone().number(NEW_STRING).srvc(service);
-        newPhone.setIsConfidential(true);
-
-        importService.createOrUpdatePhonesForService(Set.of(newPhone), service, null);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        assertEquals(0, phoneService.findAll().size());
+        assertEquals(1, phoneService.findAll().size());
+        assertEquals(EXISTING_STRING, phoneService.findAll().get(0).getNumber());
     }
 
     @Test
     @Transactional
     public void shouldNotCreateEligibilityIfConfidential() {
-        Service service = generateExistingService();
-        Eligibility eligibility = new Eligibility().eligibility(NEW_STRING).srvc(service);
+        Eligibility eligibility = new Eligibility().eligibility(NEW_STRING);
         eligibility.setIsConfidential(true);
+        Service service = generateExistingService().eligibility(eligibility);
 
-        int dbSize = eligibilityService.findAll().size();
-        importService.createOrUpdateEligibility(eligibility, service);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        assertEquals(dbSize, eligibilityService.findAll().size());
+        assertEquals(0, eligibilityService.findAll().size());
     }
 
     @Test
     @Transactional
     public void shouldNotUpdateEligibilityIfConfidential() {
-        Service service = generateExistingService();
-        Eligibility existingEligibility = new Eligibility().eligibility(EXISTING_STRING).srvc(service);
-        em.persist(existingEligibility);
-        em.flush();
-        em.refresh(service);
+        Eligibility existingEligibility = new Eligibility().eligibility(EXISTING_STRING);
+        Service service = generateExistingService().eligibility(existingEligibility);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
+        assertEquals(1, eligibilityService.findAll().size());
 
-        Eligibility newEligibility = new Eligibility().eligibility(NEW_STRING).srvc(service);
+        Eligibility newEligibility = new Eligibility().eligibility(NEW_STRING);
         newEligibility.setIsConfidential(true);
+        service.setEligibility(newEligibility);
 
-        int dbSize = eligibilityService.findAll().size();
-        Service serviceFromDb = serviceService.findWithEagerAssociations(EXISTING_EXTERNAL_ID, PROVIDER).get();
-        Eligibility updated = importService.createOrUpdateEligibility(newEligibility, serviceFromDb);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        assertEquals(dbSize, eligibilityService.findAll().size());
-        assertNull(updated);
+        assertEquals(1, eligibilityService.findAll().size());
+        assertEquals(EXISTING_STRING, eligibilityService.findAll().get(0).getEligibility());
```

Comment:
In my opinion, this is misleading because the eligibility is outdated.

Reply:
same here. I agree, but we probably have to create a separate a task for it, as it's working the same way as before changing the ImportService.